src/Examine.Lucene/Providers/BaseLuceneSearcher.cs

```diff
@@ -95,15 +98,26 @@ public override ISearchResults Search(string searchText, QueryOptions? options =
         /// </summary>
         /// <returns>Facet Config</returns>
         [Obsolete("To remove in Examine V5")]
-        public virtual FacetsConfig GetDefaultFacetConfig()
+        public virtual FacetsConfig GetDefaultFacetConfig() => new FacetsConfig();
+
+        /// <summary>
+        /// Disposes of the searcher
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
         {
-            return new FacetsConfig();
+            if (!_disposedValue)
+            {
+                _disposedValue = true;
+            }
         }
 
-        /// <inheritdoc/>
-        public virtual void Dispose()
+        /// <inheritdoc />
+        public void Dispose()
         {
-
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
```

Comment:
Align with the dispose pattern

Reply:
Looks like this would break any classes overriding public virtual void Dispose(). We should revert to keep compatibility and mark as obsolete in V5.

Reply:
@nzdev I've readded the `virtual` modifier and added an `Obsolete` attribute describing that you should override the `Dispose(bool)` instead.